### ISSUE TITLE
Ugly/benign linker warnings on macOS / AppleClang

### ISSUE THREAD

**Issue by clanmills** [Exiv2/exiv2#2116]
When I link on macOS with the M1 chip, I get thousands of warning such as:

```bash
ld: warning: direct access in function 'Exiv2::Internal::PngChunk::zlibCompress(...)'
  from file 'CMakeFiles/exiv2lib_int.dir/pngchunk_int.cpp.o'
  to global weak symbol 'typeinfo for Exiv2::BasicError<char>'
  from file 'CMakeFiles/exiv2lib.dir/error.cpp.o'
means the weak symbol cannot be overridden at runtime.
This was likely caused by different translation units being compiled with different visibility settings.
```

The fix was on redmine and I believe I opened an issue in January on GitHub to record this knowledge.  I can't find that issue report.  @nehaljwani Can you remind me how to access the read-only copy of redmine?

There is magic in cmake/mainSetup.cmake concerning visibility and it's ignored on macOS/Clang.  Enabling this on macOS doesn't make a noticeable difference.

```cmake
if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
    set(CMAKE_CXX_VISIBILITY_PRESET hidden)
    set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
endif()
```

Does anybody have a good idea?  Yup, I know I'm supposed to be the Team Exiv2 macOS expert.

**Comment by clanmills** [comment]
I found the issue on GitHub:  https://github.com/Exiv2/team/issues/2  @nehaljwani You're off the hook.  I don't need access to redmine.

The proposed fix: _Just need to add -fvisibility=hidden to EXTRA_COMPILE_FLAGS._ didn't fix it for me!

**Comment by piponazo** [comment]
@clanmills I am closing this issue, since it got fixed by merging #2141
